src/query/atomWithQuery.ts

```diff
@@ -99,25 +99,30 @@ export function atomWithQuery<
   >
   type Result = QueryObserverResult<TData, TError>
 
-  const observerAtom = atom((get) => {
-    const queryClient = getQueryClient(get)
-    const defaultedOptions = queryClient.defaultQueryOptions<
+  let observer:
```

Comment:
On second look, I think we can probably just have this in `observerAtom`.
```js
      const options =
        typeof createQuery === 'function' ? createQuery(get) : createQuery
```

Reply:
I think I tried this, but this test failed `query no-loading with keepPreviousData`
```js
const options = typeof createQuery === 'function' ? createQuery(get) : createQuery
```

Reply:
Oh, no.. Maybe that was the reason.
I'm still not sure if I understand the problem. So, we need to specify `queryKey` when we first create an observer?

Reply:
For all tests to pass, the observer can only be created once per query.

If createQuery is a function, and it's used in observerAtom, then a new observer would be created if createQuery has a dependency that changes.

I'm new to Jotai, so I don't know how we could have it read the options, without tracking its dependencies, so that only one observer is created.

Reply:
> once per query

Does it mean if the `queryKey` is the same? regardless of the other parts of the query.

Reply:
> So, we need to specify `queryKey` when we first create an observer?

Yes

Reply:
I think the problem is that the current `atomWithQuery` allows changing queryKey dynamically.
We probably shouldn't allow it.
That is going to be a breaking change in its API. If we were to add such a change, we have also other things to change.

Please let us know if you find any solution with dynamic queryKey, then we can merge it meanwhile.